### DISCUSSION TITLE
Q: Does the User need Administrator rights?

### DISCUSSION THREAD

**Discussion by JulianIsTheBestFan** [Weatherlights/Winget-AutoUpdate-Intune#14] [Q&A] [answered]
Hello,

I'm reaching out to inquire about the necessity of administrator rights for the Winget Auto Update (WAU) execution. Specifically, I'm interested in whether a user needs to have administrative privileges to operate WAU, or if it's possible to run the utility in the background without granting Admin rights.  perhaps by running it within a system context or via an alternative method?

Thanks you & greetings

Julian :)

**Comment by ak47uk** [comment]
I have deployed it using Intune to systems where the current user is a standard user, most programs will update automatically without admin rights on the currently signed-in account. However, some seem to require admin, I just had a UAC box pop-up whilst it was updating Microsoft Visual C++ 2015-2019 Redistributable (x86) - 14.38.32919.0 which has led me here to see if there is a way around this.

@Weatherlights - any ideas?

**Reply by solarplex** [reply, marked as answer]
I believe that's not something from Weatherlights but from Romanitho who is the maintainer of "Winget-AutoUpdate". Weatherlights added the Intune capabilities to the application but this repo is currently behind.

Also could be related:
https://github.com/Romanitho/Winget-AutoUpdate/issues/332

**Comment by Weatherlights** [comment]
It is possible for setup authors to define an app as user based installation and require administrative privileges to install. In this case you require administrative permissions for the user.... But it is interesting that Visual C++ is installed in the user context in the first place. Microsoft Visual C++ 2015-2019 Redistributable is installed in the computer context and does now show for when I list user installed software.

Maybe you should check how you install it in the first place here :).

**Reply by ak47uk** [reply]
Thanks, it would have been installed automatically as a prerequisite for another bit of software, which software I am not sure though...

**Comment by SynMatt** [comment]
I'm just going to use this thread for my admin rights issue.

Not sure if you’ve had this one popup @Weatherlights

I've just discovered your tool and am looking at implementing it, but it seems that RunTime needs admin rights to update.

![image](https://github.com/Weatherlights/Winget-AutoUpdate-Intune/assets/115520958/6aec0932-bec7-43f6-982d-465edf8179da)

It's a fresh install of W11 and the OS is all up-to-date.

Let me know if there is a way to update it without the user needing to be an admin :)

**Reply by Weatherlights** [reply]
Apps that are installed in the usercontext but require administrative premissions can result in this behavior. This is an issue in Winget itself and there is currently not much to work around this.

**Reply by new-it** [reply]
We have also a long list of exclusions as they all open the UAC and the users do not have permissions.
![image](https://github.com/user-attachments/assets/694acf97-6622-4896-8325-fc4cc9dea2ca)

Found this message in a MS technet articel:
![image](https://github.com/user-attachments/assets/93d3b966-bd3d-4b75-bad6-0744f9f0d898)

I'm a bit confused which it's not prevent when the WAU is installed as system and runs it as system. Is there really no option to prevent the UAC directly and skip the update. We have more then 1400 Apps in the env. and we cannot test everything.
